### DISCUSSION TITLE
Use non-unique fields in addition to unique fields in delete query

### DISCUSSION THREAD

**Discussion by aheidelberg** [prisma/prisma#13111] [Q&A] [answered]
```prisma
model User {
  id             String            @id @default(cuid())
  linkedAccounts LinkedAccount[]
}

model LinkedAccount {
  id          Int          @id @default(autoincrement())
  user        User         @relation(fields: [userId], references: [id])
  userId      String
}
```

```typescript
await ctx.prisma?.linkedAccount.delete({
  where: {
    id: linkedAccountId, // this comes from frontend
    userId: authenticatedUserId, // this comes from the session / backend and is authenticated
  }
});
```

I'm trying to delete an entity that's linked to a user and want to validate that the user is actually allowed to delete it.
My idea was to just add the `userId` to the `where` clause but that's not possible because only unique values are allowed in there and `userId` is not unique since this is a one-to-many relationship.

I don't really want to do a separate query before the delete query just to confirm the `linkedAccountId` from the frontend belongs to the authenticated user when it could be possible in one single query.

Is there any way around this?

EDIT: Just realized I can just use `deleteMany`. Feels a bit weird because there will only ever be one entity that is going to be deleted. Is there any disadvantage to doing it that way or a better way?

**Comment by nurul3101** [comment, marked as answer]
Hey @aheidelberg 👋 

As you mentioned you can use the `deleteMany` query to pass non-unique fields. 
Other alternatives are to use [Raw Query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access), or to add multi-field index. 

We have a Feature Request: #10376 which discusses whereUnique constraints in single object lookups, if you could leave a 👍 to the request, it would help our engineering and product team in prioritising it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
